docker/register.sh

```diff
@@ -36,60 +40,57 @@ register_operator_with_avs() {
     if [ -z "$avs_directory_address" ]; then
-        echo "Error: Failed to read AVSDirectory from /root/.nodes/avs_deploy.json"
+        echo "Error: Failed to get AVSDirectory from ${WAVS_SERVICE_MANAGER_ADDRESS} avsDirectory()"
         exit 1
     fi
     # Generate a random salt (32 bytes)
     local salt=$(openssl rand -hex 32)
-    
+
     # Calculate expiry (current time + 1 hour)
     local expiry=$(($(date +%s) + 3600))
-    
-    local digest_hash=$(cast call "$avs_directory_address" "calculateOperatorAVSRegistrationDigestHash(address,address,bytes32,uint256)" "$public_key" "$service_manager_address" "$salt" "$expiry" --rpc-url "$LOCAL_ETHEREUM_RPC_URL")
+
+    local digest_hash=$(cast call "$avs_directory_address" "calculateOperatorAVSRegistrationDigestHash(address,address,bytes32,uint256)" "$public_key" "$WAVS_SERVICE_MANAGER_ADDRESS" "$salt" "$expiry" --rpc-url "$LOCAL_ETHEREUM_RPC_URL")
     # Remove 0x prefix from digest hash if present
     digest_hash=${digest_hash#0x}
     # Sign the digest hash with the private key
     local signature=$(cast wallet sign $digest_hash --no-hash --private-key "$private_key")
-    
-    # Register the operator with the signature
-    echo "Registering operator with signature..."
-    cast c --trace "$stake_registry_address" \
-        "registerOperatorWithSignature((bytes,bytes32,uint256),address)" \
-        "($signature,$salt,$expiry)" "$public_key" \
-        --private-key "$private_key" \
-        --rpc-url "$LOCAL_ETHEREUM_RPC_URL" \
-    
-    cast send "$stake_registry_address" \
-        "registerOperatorWithSignature((bytes,bytes32,uint256),address)" \
-        "($signature,$salt,$expiry)" "$public_key" \
-        --private-key "$private_key" \
-        --rpc-url "$LOCAL_ETHEREUM_RPC_URL" 
-    if [ $? -eq 0 ]; then
-        echo "Successfully registered operator $public_key with AVS"
+
+    local operatorRegistered=$(cast call "$STAKE_REGISTRY_ADDRESS" "operatorRegistered(address)(bool)" "$public_key" --rpc-url "$LOCAL_ETHEREUM_RPC_URL")
+    if [ "$operatorRegistered" = "false" ]; then
+        echo "Registering operator with signature..."
+        cast c "$STAKE_REGISTRY_ADDRESS" \
+            "registerOperatorWithSignature((bytes,bytes32,uint256),address)" \
+            "($signature,$salt,$expiry)" "$public_key" \
+            --private-key "$private_key" \
+            --rpc-url "$LOCAL_ETHEREUM_RPC_URL" \
+
+        cast send "$STAKE_REGISTRY_ADDRESS" \
+            "registerOperatorWithSignature((bytes,bytes32,uint256),address)" \
+            "($signature,$salt,$expiry)" "$public_key" \
+            --private-key "$private_key" \
+            --rpc-url "$LOCAL_ETHEREUM_RPC_URL"
+        if [ $? -eq 0 ]; then
+            echo "Successfully registered operator $public_key with AVS"
+        else
+            echo "Error: Failed to register operator with AVS"
```

Comment:
This will never be hit, as we have `set -o errexit` on top, which means the script will terminate as soon as any line returns a non-zero error code.

You could do something like `cast send ... || (echo "Error: Failed to register" && exit 1)` to have the same effect